internal/plugin/server.go

```diff
@@ -361,26 +362,31 @@ func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*plu
 // This includes per-resource pipe and log directories as well as a global daemon-specific shm
 // and assumes that an MPS control daemon has already been started.
 func (plugin NvidiaDevicePlugin) updateResponseForMPS(response *pluginapi.ContainerAllocateResponse) {
-	pipeDir := filepath.Join("/mps", string(plugin.rm.Resource()), "pipe")
-	response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = pipeDir
+	// TODO: We should check that the deviceIDs are shared using MPS.
+
+	// TODO: We use the Daemon here just to construct the pipe and log dirs for the specified resource.
+	containerMpsDaemon := mps.NewDaemon(plugin.rm, "/mps")
+
+	containerPipeDir := containerMpsDaemon.PipeDir()
+	containerLogDir := containerMpsDaemon.LogDir()
+
+	response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = containerPipeDir
+	response.Envs["CUDA_MPS_LOG_DIRECTORY"] = containerLogDir
```

Comment:
Let's clean this up before merging.

Reply:
I've reworked this, reusing the `mpsDaemon` from plugin construction and adding an `mps.Root` type.

Reply:
Did you forget to push?

Reply:
SHould be updated now.

Reply:
I still don't see the update

Reply:
🤦  let's blame the flight.